src/wasm/wasm-emscripten.cpp

```diff
@@ -230,10 +230,14 @@ Function* EmscriptenGlueGenerator::generateAssignGOTEntriesFunction() {
   }
 
   for (Global* g : got_entries_func) {
-    Name getter(std::string("fp$") + g->base.c_str());
-    if (auto* f = wasm.getFunctionOrNull(g->base)) {
-      getter.set((getter.c_str() + std::string("$") + getSig(f)).c_str(), false);
+    auto* f = wasm.getFunctionOrNull(g->base);
+    if (!f) {
+      auto* ex = wasm.getExportOrNull(g->base);
```

Comment:
Can just be `getExport` here, which also checks that the export is non-null (something something undefined behavior)